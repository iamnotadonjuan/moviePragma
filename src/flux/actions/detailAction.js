import { INIT_FAVORITE, SAVE_RATING } from '../actionTypes'

const initFavorite = (data) => {
  return async dispatch => {
    const favorite = [{
      star_1: false,
      star_2: false,
      star_3: false,
      star_4: false,
      star_5: false
    }
    ]
    dispatch(_initFavorite(favorite))
  }
}

const _initFavorite = (data) => {
  return {
    type: INIT_FAVORITE,
    payload: { data }
  }
}

const saveRating = (data) => {
  return async dispatch => {

    if (data.count > 1) {
      let i;
      for (i = 1; i <= data.count; i++) {
        if (data.value == true && data.count < 5) {
          for (i = data.count; i <= 5; i++) {
            dataCopy = {
              position: 'star_' + i,
              value: data.value
            }
            dispatch(_saveRating(dataCopy))
          }
        } 

        dataCopy = {
          position: 'star_' + i,
          value: data.value
        }
        dispatch(_saveRating(dataCopy))
      }
    }
    else {
      if (data.value == true) {
        _initStar(dispatch, data)
      } else {
        dispatch(_saveRating(data))
      }
    }
  }
}

const _saveRating = (data) => {
  return {
    type: SAVE_RATING,
    payload: { data }
  }
}

const _initStar = (dispatch,data) => {
  for (i = data.count; i <= 5; i++) {
    dataCopy = {
      position: 'star_' + i,
      value: data.value
    }
    dispatch(_saveRating(dataCopy))
  }
}


export { initFavorite, saveRating }