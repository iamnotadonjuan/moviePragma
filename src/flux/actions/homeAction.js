import { MOVIE_LIST } from '../actionTypes'
import API from '../../services/api'

const listMovie =  (data) => {
  return async dispatch => {
    const List = await API.get(data.limit, '3D', data.url)
    dispatch(_listMovie(List))
  }
}

const _listMovie = (data) => {
  return {
    type: MOVIE_LIST,
    payload: { data }
  }
}


export { listMovie }