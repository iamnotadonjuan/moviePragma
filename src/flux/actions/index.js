export { login, initSession, saveUserInfo } from './loginAction'