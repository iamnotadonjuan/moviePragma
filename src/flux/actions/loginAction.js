import { LOGIN_ACTION, INIT_SESSION } from '../actionTypes'

const login = (data) => {
  return dispatch => {
    dispatch(_login(user))
  }
}

const _login = (data) => {
  return {
    type: LOGIN_ACTION,
    paylod: { data }
  }
}

const initSession = data => {
    return dispatch => {
      dispatch(_initSession(data))
    }
}

const _initSession = (data) => {
  return {
    type: INIT_SESSION,
    payload: { data }
  }
}

export { login, initSession }