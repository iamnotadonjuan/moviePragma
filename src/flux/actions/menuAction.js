import { CLOSE_MENU } from '../actionTypes'

const closeMenu =  (data) => {
  return async dispatch => {
    dispatch(_closeMenu(data))
  }
}

const _closeMenu = (data) => {
  return {
    type: CLOSE_MENU,
    payload: { data }
  }
}


export { closeMenu }