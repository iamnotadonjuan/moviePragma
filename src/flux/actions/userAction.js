import { SAVE_USER } from '../actionTypes'
import {
  AsyncStorage
} from 'react-native'

const saveUserInfo = (data) => {
  return async dispatch => {
    try {
      await AsyncStorage.setItem('currentUser', JSON.stringify(data))
      dispatch(_saveUserInfo(data))
    } catch (error) {
      console.log(error)
    }
  }
}

const retrieveUserInfo = async (data) => {
  try {
    return await AsyncStorage.getItem('currentUser')
  } catch (error) {
    console.log(error)
  }
}

const _saveUserInfo = (data) => {
  return {
    type: SAVE_USER,
    payload: { data }
  }
}

export { saveUserInfo, retrieveUserInfo }
