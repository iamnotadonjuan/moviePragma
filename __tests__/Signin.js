import 'react-native'
import React from 'react'
import SignInController from '../src/controllers/SignInController';
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import thunk from "redux-thunk";
import configureStore from 'redux-mock-store'
import { saveUserInfo, retrieveUserInfo } from '../src/flux/actions/userAction'

const buildStore = configureStore([thunk]);

configure({ adapter: new Adapter() })
describe('Testing SignIn component', () => {
  it('SignIn renderers', () => {
    const wrapper = shallow(
      <SignInController />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Testing requests', () => {
  let store;
  const initialState = {
    isLogged: false
  }
  beforeEach(() => {
    store = buildStore(initialState);
  });
  it("Init session", async () => {
    const data = { displayName: 'test', email: 'test@test.com', uuid: 1002 }
    await store.dispatch(saveUserInfo(data))
    expect(store.getActions()).toMatchSnapshot()
  });

}
)