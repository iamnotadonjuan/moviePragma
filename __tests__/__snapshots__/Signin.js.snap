// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignIn component SignIn renderers 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Testing requests Init session 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Object {
        "displayName": "test",
        "email": "test@test.com",
        "uuid": 1002,
      },
    },
    "type": "save_user",
  },
]
`;
